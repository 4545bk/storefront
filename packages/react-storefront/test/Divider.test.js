/**
 * @license
 * Copyright Â© 2017-2018 Moov Corporation.  All rights reserved.
 */
import React from 'react'
import { mount } from 'enzyme'
import Divider from '../src/Divider'

describe('Divider', () => {

  it('defaults to horizontal', () => {
    const component = (
      <Divider/>
    )

    expect(mount(component)).toMatchSnapshot()
  })

  it('renders a vertial divider', () => {
    const component = (
      <Divider vertical/>
    )

    expect(mount(component)).toMatchSnapshot()
  })
  
  it('renders a horizontal divider', () => {
    const component = (
      <Divider horizontal/>
    )

    expect(mount(component)).toMatchSnapshot()
  })

  it('accepts a className', () => {
    const component = (
      <Divider className="myClass"/>
    )

    expect(mount(component)).toMatchSnapshot()
  })

  it('passes styles to the underlying div', () => {
    const component = (
      <Divider style={{ color: 'red' }}/>
    )

    expect(mount(component)).toMatchSnapshot()
  })
})