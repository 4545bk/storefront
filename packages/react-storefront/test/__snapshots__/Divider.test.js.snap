// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Divider accepts a className 1`] = `
<WithStyles(Divider) className="myClass">
  <Divider className="myClass" classes={{...}}>
    <div className="myClass RSFDivider-root-1 RSFDivider-horizontal-3" />
  </Divider>
</WithStyles(Divider)>
`;

exports[`Divider defaults to horizontal 1`] = `
<WithStyles(Divider)>
  <Divider classes={{...}}>
    <div className="RSFDivider-root-1 RSFDivider-horizontal-3" />
  </Divider>
</WithStyles(Divider)>
`;

exports[`Divider passes styles to the underlying div 1`] = `
<WithStyles(Divider) style={{...}}>
  <Divider style={{...}} classes={{...}}>
    <div className="RSFDivider-root-1 RSFDivider-horizontal-3" style={{...}} />
  </Divider>
</WithStyles(Divider)>
`;

exports[`Divider renders a horizontal divider 1`] = `
<WithStyles(Divider) horizontal={true}>
  <Divider horizontal={true} classes={{...}}>
    <div className="RSFDivider-root-1 RSFDivider-horizontal-3" />
  </Divider>
</WithStyles(Divider)>
`;

exports[`Divider renders a vertial divider 1`] = `
<WithStyles(Divider) vertical={true}>
  <Divider vertical={true} classes={{...}}>
    <div className="RSFDivider-root-1 RSFDivider-vertical-2" />
  </Divider>
</WithStyles(Divider)>
`;
